packages/dapp/src/components/Account/LoadUser.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from "react";
+import { getApolloSession } from "@joincivil/utils";
+import gql from "graphql-tag";
+import { Query } from "react-apollo";
+
+const userQuery = gql`
+  query {
+    currentUser {
+      uid
+      email
+      ethAddress
+      quizPayload
+      quizStatus
+    }
+  }
+`;
+
+export interface LoadUserChildrenProps {
+  user: any;
+  loading: boolean;
+}
+
+export interface LoadUserProps {
+  children(props: any): any;
+}
+
+export const LoadUser: React.SFC<LoadUserProps> = props => {
+  const auth = getApolloSession();
+
+  const renderFunction = props.children;
+  const hasAuthToken = auth && auth.token;
+
+  if (!hasAuthToken) {
+    return <>{renderFunction({ user: null, loading: false })}</>;
+  }
+
+  return (
+    <Query query={userQuery}>
+      {({ loading, error, data }) => {
+        if (loading || error) {
```

Comment:
If you haven't already you should double check that this handles the logged-out state (which will cause this request to error) in the way you expect - unless you're only using this component in routes that require authentication?